packages/iov-cosmos/src/encode.spec.ts

```diff
@@ -0,0 +1,322 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import {
+  Address,
+  Algorithm,
+  Amount,
+  ChainId,
+  Identity,
+  Nonce,
+  PubkeyBytes,
+  SignatureBytes,
+  TokenTicker,
+} from "@iov/bcp";
+import { Encoding } from "@iov/encoding";
+
+import {
+  buildSignedTx,
+  buildUnsignedTx,
+  encodeAmount,
+  encodeFee,
+  encodeFullSignature,
+  encodePubkey,
+} from "./encode";
+
+const { fromBase64 } = Encoding;
+
+describe("encode", () => {
+  // https://rpc.cosmos.network:26657/tx?hash=0x2268EB5AB730B45F8426078827BB5BB49819CE2B0D74B2C1D191070BADB379F1&prove=true
+  const defaultPubkey = {
+    algo: Algorithm.Secp256k1,
+    data: fromBase64("AtQaCqFnshaZQp6rIkvAPyzThvCvXSDO+9AzbxVErqJP") as PubkeyBytes,
+  };
+  const defaultCreator: Identity = {
+    chainId: "not-used" as ChainId,
+    pubkey: defaultPubkey,
+  };
+  const defaultSender = "cosmos1h806c7khnvmjlywdrkdgk2vrayy2mmvf9rxk2r" as Address;
+  const defaultRecipient = "cosmos1z7g5w84ynmjyg0kqpahdjqpj7yq34v3suckp0e" as Address;
+  const defaultAmount: Amount = {
+    quantity: "11657995",
+    fractionalDigits: 9,
+    tokenTicker: "vatom" as TokenTicker,
```

Comment:
Why do we use uatom and vatom?

Reply:
I set up our local test chain with "vatom" to make it clear it wasn't mainnet, and "uatom" comes from real life transactions from the CosmosHub chain that are used in some test cases. I didn't think too rigorously about when to use which.